File: solutions/0001-two-sum/solution.py
class Solution:
        def twoSum(self, numbers: List[int], target: int) -> List[int]:
            seen = {}

            for i, num in enumerate(numbers):
                fark = target - num
                if fark in seen:
                    return [seen[fark], i]

                seen[num] = i

File: solutions/0001-two-sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> hafiza;

        for (int i = 0; i < nums.size(); i++){
            int suAnkiSayi = nums[i];
            int arananFark = target - suAnkiSayi;

            if (hafiza.find(arananFark) != hafiza.end()){
                return {hafiza[arananFark], i};
            }
            hafiza[suAnkiSayi] = i;
        }
        return {};
    }
};
